src/main/java/com/thealgorithms/backtracking/combinationSum.java

```diff
@@ -0,0 +1,75 @@
+package com.thealgorithms.backtracking;
+
+import java.util.*;
+
+/**
+ * Java program for solving the Combination Sum problem.
+ * Author: Yash Kesharwani
+ * File: CombinationSum.java
+ * Comments: This program finds all combinations of candidates that sum to the target.
+ */
+public class combinationSum {
+    /**
+     * Recursively find combinations that sum to the target.
+     *
+     * @param res       The list to store valid combinations.
+     * @param candidates The array of candidate numbers.
+     * @param target    The target sum to achieve.
+     * @param sum1      The current sum during the recursion.
+     * @param start     The starting index for candidates.
+     * @param list1     The current combination being constructed.
+     */
+    public void check(List<List<Integer>> res, int[] candidates, int target, int sum1, int start, List<Integer> list1) {
```

Comment:
Is it the same as https://github.com/TheAlgorithms/Java/blob/574138c7a35351a0837bb4bd56e2eb295064b690/src/main/java/com/thealgorithms/dynamicprogramming/Sum_Of_Subset.java#L3?

Reply:
No both of these questions are different @siriak